### DISCUSSION TITLE
Scope of ENV.deparallelize

### DISCUSSION THREAD

**Discussion by lelegard** [Homebrew/discussions#3076] [Writing Formulae/Casks] [answered]
Hi,

In the [formula cookbook](https://docs.brew.sh/Formula-Cookbook), we have this example:
~~~
ENV.deparallelize
system "make"  # separate make and make install steps
system "make", "install"
~~~

What is the scope of `ENV.deparallelize` in terms of sequence? Does it apply to the entire `def install` section or only starting at the line it is mentioned?

In short, following a possible race condition in the `install -d` shell command on macOS ([discussion here](https://github.com/Homebrew/homebrew-core/pull/96790#event-6231500249)), I would like to make sure that the build command is parallelized (400 kloc of C++) but the `make install` command should not be.

Would the following solve this question?
~~~
system "make"
ENV.deparallelize
system "make", "install"
~~~

**Comment by Bo98** [comment, marked as answer]
It is starting at the `ENV.deparallelize` line, so yes that would work.

It isn't necessary for to your example, but if you wanted to _restore_ parallelisation later on then there's also this variant:

```rb
ENV.deparallelize do
  system "make", "install"
end
# Parellelisation active again here
```

**Reply by lelegard** [reply]
Thanks
